packages/flutter/lib/src/widgets/async.dart

```diff
@@ -632,3 +789,39 @@ class _FutureBuilderState<T> extends State<FutureBuilder<T>> {
     _activeCallbackIdentity = null;
   }
 }
+
+/// Class that allows callers to reference instances of [Type] that would
+/// otherwise not be valid expressions.
+///
+/// Generic types and the `void` type are not usable as Dart expressions, so
+/// the following statements are not legal and all yield compile-time errors:
+///
+/// ```dart
+/// if (type == List<int>) print('msg');
+/// if (type == void) print('msg');
+/// Type type = List<int>;
+/// ```
+///
+/// This class allows callers to get handles on such types, like so:
+///
+/// ```dart
+/// if (type == const _TypeLiteral<List<int>>().type) print('msg');
+/// if (type == const _TypeLiteral<void>().type) print('msg');
+/// Type type = const _TypeLiteral<List<int>>().type;
+/// ```
+class _TypeLiteral<T> {
+  /// Creates a new [_TypeLiteral].
+  const _TypeLiteral();
+
+  /// Returns whether the specified type represents a "void" type.
+  ///
+  /// Currently in the Dart language, both the `Null` type and the `void` type
+  /// represents "void" types (types that contain no value).
+  static bool isVoidType(Type type) {
+    return type == const _TypeLiteral<Null>().type  // ignore: prefer_void_to_null
```

Comment:
i wouldn't accept Null here. If it's a `Future<Null>` and it completes (with null) then I do have data: null.